test/preview/dataclasses/test_document.py

```diff
@@ -75,71 +83,83 @@ def test_init_document_same_meta_as_main_fields():
     This is forbidden to prevent later issues with `Document.flatten()`
     """
     with pytest.raises(ValueError, match="score"):
-        Document(content="test content", metadata={"score": "10/10"})
+        Document(text="test text", metadata={"score": "10/10"})
 
 
 @pytest.mark.unit
 def test_basic_equality_type_mismatch():
-    doc = Document(content="test content")
-    assert doc != "test content"
-
-
-@pytest.mark.unit
-def test_simple_table_document_equality():
-    doc1 = Document(content=pd.DataFrame([1, 2]), content_type="table")
-    doc2 = Document(content=pd.DataFrame([1, 2]), content_type="table")
+    doc = Document(text="test text")
+    assert doc != "test text"
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "doc1,doc2",
+    [
+        [Document(text="test text"), Document(text="test text")],
+        [Document(text="test text", score=100), Document(text="test text", score=100)],
+        [
+            Document(text="test text", embedding=np.array([10, 10])),
+            Document(text="test text", embedding=np.array([10, 10])),
+        ],
+        [
+            Document(text="test text", metadata={"value": 1, "another": "value"}),
+            Document(text="test text", metadata={"value": 1, "another": "value"}),
+        ],
+        [
+            Document(text="test text", metadata={"value": {"another": "value"}}),
+            Document(text="test text", metadata={"value": {"another": "value"}}),
+        ],
+        [Document(array=np.array([1, 2])), Document(array=np.array([1, 2]))],
+        [Document(dataframe=pd.DataFrame([1, 2])), Document(dataframe=pd.DataFrame([1, 2]))],
+        [Document(blob=b"some bytes"), Document(blob=b"some bytes")],
+    ],
+)
+def test_simple_equality(doc1, doc2):
     assert doc1 == doc2
 
 
 @pytest.mark.unit
-def test_simple_image_document_equality():
-    doc1 = Document(content=Path(__file__).parent / "test_files" / "apple.jpg", content_type="image")
-    doc2 = Document(content=Path(__file__).parent / "test_files" / "apple.jpg", content_type="image")
-    assert doc1 == doc2
-
-
-@pytest.mark.unit
-def test_equality_with_embeddings():
-    doc1 = Document(content="test content", embedding=np.array([10, 10]))
-    doc2 = Document(content="test content", embedding=np.array([10, 10]))
-    assert doc1 == doc2
-
-
-@pytest.mark.unit
-def test_equality_with_embeddings_shape_check():
-    doc1 = Document(content="test content", embedding=np.array([10, 10]))
-    doc2 = Document(content="test content", embedding=np.array([[[10, 10]]]))
+@pytest.mark.parametrize(
+    "doc1,doc2",
+    [
+        [Document(text="test text", score=100), Document(text="test text")],
+        [Document(text="test text", score=100), Document(text="test text", score=101)],
```

Comment:
I'm not sure if having a different score/no score should have an effect on Document equality...

Reply:
Open to suggestions on this point :blush: No problem changing this

Reply:
I ended up simplifying drastically how equality works for documents, so the problem is gone